features/prog.go

```diff
@@ -92,7 +93,16 @@ var haveProgramTypeMatrix = internal.FeatureMatrix[ebpf.ProgramType]{
 			})
 		},
 	},
-	// ebpf.Tracing:               {Version: "5.5"},
+	ebpf.Tracing: {
+		Version: "5.5",
+		Fn: func() error {
+			return probeProgram(&ebpf.ProgramSpec{
+				Type:       ebpf.Tracing,
+				AttachType: ebpf.AttachTraceFEntry,
+				AttachTo:   "inet_dgram_connect",
```

Comment:
I'm hesitant to hardcode ksyms outside of the test suite, even the oldest of subsystems are sometimes subject to change if there's a need to modernize/refactor. Since fentry relies on `findTargetInKernel()`, we can do the following without any real cost:
```go
func() error {
	s, err := btf.LoadKernelSpec()
	if err != nil {
		return err
	}

	i := s.Iterate()
	for i.Next() {
		// Find the first named kernel function.
		f, ok := i.Type.(*btf.Func)
		if !ok || f.Name == "" {
			continue
		}
		return probeProgram(&ebpf.ProgramSpec{
			Type:       ebpf.Tracing,
			AttachType: ebpf.AttachTraceFEntry,
			AttachTo:   f.Name,
		})
	}

	return errors.New("no suitable kernel function for fentry")
},
```

Reply:
Some functions may not be traceable, I think this may lead to more harm than good.

Reply:
True, I decided to gather some data. Aside from the obvious issue described in #894, there are currently two clear limitations to Tracing programs. Fortunately, the BTF verifier is being really helpful:

1. `invalid argument: The function kthread_create_worker_on_cpu with variable args is unsupported.`
```c
struct kthread_worker *
kthread_create_worker_on_cpu(int cpu, unsigned int flags,
			         const char namefmt[], ...)
```
So, anything with varargs currently isn't supported.

2. `invalid argument: The function __copy_xstate_to_uabi_buf arg0 type STRUCT is unsupported.`
```c
void __copy_xstate_to_uabi_buf(struct membuf to, struct fpstate *fpstate,
			           u32 pkru_val, enum xstate_copy_mode copy_mode)
```
Any function taking by-value struct args larger than 16 bytes is not supported.

This seems to be governed by [btf_distill_func_proto()](https://elixir.bootlin.com/linux/v6.1.1/source/kernel/bpf/btf.c#L5917).

With that off the table, there's still the option of retrying with up to N funcs if we're not hardcoding a symbol.

Reply:
I ranked all Funcs containing `bpf` in my kernel's BTF blob by length and found a strong candidate for hardcoding as an fentry probe target: [`bpf_init`](https://elixir.bootlin.com/linux/v6.1.1/source/kernel/bpf/inode.c#L806). It's been part of the kernel since 4.4 (https://github.com/torvalds/linux/commit/b2197755b263), has been completely untouched, and has a simple signature. For probing a BPF program type, it makes sense to target a symbol from the BPF subsystem. @borkmann mentioned that this is unlikely to change.

I'll push a fixup to quickly put this through CI and go for merge.